pkg/client/oci/oci.go

```diff
@@ -0,0 +1,302 @@
+package oci
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"time"
+
+	"github.com/juju/errors"
+	"k8s.io/klog"
+
+	"github.com/bitnami-labs/charts-syncer/api"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/types"
+	"github.com/bitnami-labs/charts-syncer/pkg/helmcli"
+	"github.com/bitnami-labs/charts-syncer/pkg/utils"
+)
+
+const (
+	getTimeout = 1 * time.Minute
+	// HelmChartConfigMediaType is the reserved media type for the Helm chart manifest config
+	HelmChartConfigMediaType = "application/vnd.cncf.helm.config.v1+json"
+
+	// HelmChartContentLayerMediaType is the reserved media type for Helm chart package content
+	HelmChartContentLayerMediaType = "application/tar+gzip"
+)
+
+// Repo allows to operate a chart repository.
+type Repo struct {
+	url      *url.URL
+	username string
+	password string
+}
+
+// Tags contains the tags for a specific OCI artifact
+type Tags struct {
+	Name string
+	Tags []string
+}
+
+// LayerInfo contains info about a OCI layer
+type LayerInfo struct {
+	MediaType string
+	Digest    string
+	Size      int
+}
+
+// TagManifest contains info about the published TAG in OCI
+type TagManifest struct {
+	SchemaVersion int
+	Config        LayerInfo
+	Layers        []LayerInfo
+}
```

Comment:
have you checked the [upstream OCI schema](https://godoc.org/github.com/moby/moby/vendor/github.com/docker/distribution/manifest/ocischema#Manifest)?

Reply:
using this instead:

- https://github.com/opencontainers/image-spec/blob/master/specs-go/v1/manifest.go